elements/rh-pagination/rh-pagination.ts

```diff
@@ -0,0 +1,215 @@
+   * the list items.
+   */
+  @property({ reflect: true }) overflow: 'start' | 'end' | 'both' | null = null;
+
+  /** Accessible label for the 'nav' element */
+  @property() label = 'Page navigation';
+
+  /** Accessible label for the 'first page' button */
+  @property({ attribute: 'label-first' }) labelFirst = 'first page';
+
+  /** Accessible label for the 'previous page' button */
+  @property({ attribute: 'label-previous' }) labelPrevious = 'previous page';
+
+  /** Accessible label for the 'next page' button */
+  @property({ attribute: 'label-next' }) labelNext = 'next page';
+
+  /** Accessible label for the 'last page' button */
+  @property({ attribute: 'label-last' }) labelLast = 'last page';
+
+  #mo = new MutationObserver(() => this.#update());
+  #screen = new ScreenSizeController(this);
+  #logger = new Logger(this);
+
+  #ol = this.querySelector('ol');
+  #links = this.#ol?.querySelectorAll<HTMLAnchorElement>('li a');
+
+  #currentIndex = -1;
+  #currentLink = this.#getCurrentLink();
+  #firstLink: HTMLAnchorElement | null = null;
+  #lastLink: HTMLAnchorElement | null = null;
+  #nextLink: HTMLAnchorElement | null = null;
+  #prevLink: HTMLAnchorElement | null = null;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.#mo.observe(this, { childList: true, subtree: true });
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.#mo.disconnect();
+  }
+
+  update(changed: PropertyValues<this>): void {
+    this.#update();
+    super.update(changed);
+  }
+
+  render() {
+    const { mobile, size } = this.#screen;
+    const { label, labelFirst, labelPrevious, labelNext, labelLast } = this;
+    const firstHref = this.#currentLink === this.#firstLink ? undefined : this.#firstLink?.href;
+    const prevHref = this.#prevLink?.href;
+    const nextHref = this.#nextLink?.href;
+    const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
+    return html`
+      <div id="container" class=${classMap({ mobile, [size as string]: true })}>
+        <a id="first" class="stepper" href=${ifDefined(firstHref)} ?inert=${!firstHref} aria-label=${labelFirst}>${L2}</a>
+        <a id="prev" class="stepper" href=${ifDefined(prevHref)} ?inert=${!prevHref} aria-label=${labelPrevious}>${L1}</a>
+
+        <nav ?hidden=${mobile} ?inert=${mobile} aria-label=${label}>
+          <slot></slot>
+        </nav>
+
+        <a id="next" class="stepper" href=${ifDefined(nextHref)} ?inert=${!nextHref} aria-label=${labelNext}>${L1}</a>
+        <a id="last" class="stepper" href=${ifDefined(lastHref)} ?inert=${!lastHref} aria-label=${labelLast}>${L2}</a>
+
+        ${!mobile ? '' : html`
+        <div id="numeric">
+          <span id="go-to-page">
+            <slot name="go-to-page">Go to page</slot>
+          </span>
+          <input inputmode="numeric" aria-labelledby="go-to-page" value=${(this.#currentIndex ?? 0) + 1} />
```

Comment:
Does it make sense to also have this input as type="number" and do a min and max range?

```Suggestion
          <input type="number" inputmode="numeric" aria-labelledby="go-to-page" min="1" max="${this.#links?.length}" value=${(this.#currentIndex ?? 0) + 1}  />
```

I would presume this should also trigger  `onchange` when the value is changed?

Reply:
we don't want `type=number` since we provide our own spinners. `inputmode=numeric` suffices for the design spec. adding validations is a good idea though. I added the min max and required attrs, although they don't actually do anything, and we still need to manually validate. Still, i think there's value in exposing intent and for a few bytes of HTML it's worth the cost.